addons/l10n_ar/models/res_partner.py

```diff
@@ -52,6 +53,18 @@ def _compute_l10n_ar_vat(self):
         remaining = self - recs_ar_vat
         remaining.l10n_ar_vat = False
 
+    @api.depends('l10n_latam_identification_type_id', 'l10n_ar_vat')
+    def _compute_l10n_ar_is_company(self):
+        "True if partner is considered a company in Argentina, based on Identification Type and CUIT prefix."
+        for rec in self:
+            afip_code = rec.l10n_latam_identification_type_id.l10n_ar_afip_code
+            prefix = (rec.l10n_ar_vat or '')[:2]
+
+            if afip_code == '80' and prefix in ('30', '33', '34', '51', '55'):  # CUIT
```

Comment:
This logic does not make much sense as we need it for foreigners, not internally in the country.  You could simply check the is_vat field on the identification type (and check that the vat field is set) and work from there, but e.g. l10n_sa uses the additional identification and not the vat field.  @vbe-odoo  @jjscarafia @zaoral

Reply:
Hello @jco-odoo, thanks for the ping on this one!
We've checked with @jjscarafia and we agree with you that domestically, we can avoid differentiating, but only for foreign transactions. For that purpose, we could use the is_vat as you suggested.

We left something along those lines here:
https://github.com/odoo/enterprise/pull/86089#discussion_r2128977870

Thanks!

Reply:
Thanks @vbe-odoo @jco-odoo 
I've make this PR's with a proofe of concepto of what could be changed in l10n_ar to remove usage of is_company:
* https://github.com/odoo/odoo/pull/222957
* https://github.com/odoo/enterprise/pull/92361